permission_handler/android/src/main/java/com/baseflow/permissionhandler/PermissionManager.java

```diff
@@ -271,21 +292,38 @@ private int checkNotificationPermissionStatus(Context context) {
             this.callback = callback;
         }
 
+        @VisibleForTesting
+        ActivityResultListener(RequestPermissionsSuccessCallback callback, Context context) {
+            this.callback = callback;
+            this.context = context;
+        }
+
         @Override
         public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
-            if(alreadyCalled || requestCode != PermissionConstants.PERMISSION_CODE_IGNORE_BATTERY_OPTIMIZATIONS) {
-                return false;
+            if (requestCode == PermissionConstants.PERMISSION_CODE_IGNORE_BATTERY_OPTIMIZATIONS) {
+                if (alreadyCalled)
+                    return false;
+                alreadyCalled = true;
+                final int status = resultCode == Activity.RESULT_OK ? PermissionConstants.PERMISSION_STATUS_GRANTED
+                        : PermissionConstants.PERMISSION_STATUS_DENIED;
+
+                HashMap<Integer, Integer> results = new HashMap<>();
+                results.put(PermissionConstants.PERMISSION_GROUP_IGNORE_BATTERY_OPTIMIZATIONS, status);
+                callback.onSuccess(results);
+                return true;
+            } else if (requestCode == PermissionConstants.PERMISSION_GROUP_SYSTEM_ALERT_WINDOW && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                if (alreadySystemAlertWindowCallbackCalled)
+                    return false;
+                alreadySystemAlertWindowCallbackCalled = true;
```

Comment:
As mentioned before, we should improve the registration of the activity listener so we don't need this construct and the `onActivityResult` method is only called once.